plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/ray_launcher.py

```diff
@@ -0,0 +1,197 @@
+    ) -> Sequence[JobReturn]:
+        setup_globals()
+        assert self.config is not None
+        assert self.config_loader is not None
+        assert self.task_function is not None
+
+        # based on mode, populate self.ray_yaml_path
+        ray_yaml_path = self._save_ray_config()
+
+        # Add remote dir to the search path. TODO see if there's a better way to do this
+        remote_script_dir = self._get_remote_dir(ray_yaml_path)
+
+        configure_log(self.config.hydra.hydra_logging, self.config.hydra.verbose)
+        sweep_dir = Path(str(self.config.hydra.sweep.dir))
+        sweep_dir.mkdir(parents=True, exist_ok=True)
+        log.info("Ray Launcher is launching {} jobs".format(len(job_overrides)))
+        log.info("Sweep output dir : {}".format(sweep_dir))
+
+        temp_dir_job_dump = tempfile.mkdtemp()
+
+        for idx, overrides in enumerate(job_overrides):
+            idx = initial_job_idx + idx
+            log.info("\t#{} : {}".format(idx, " ".join(filter_overrides(overrides))))
+            sweep_config = self.config_loader.load_sweep_config(
+                self.config, list(overrides)
+            )
+            with open_dict(sweep_config):
+                # This typically coming from the underlying scheduler (SLURM_JOB_ID for instance)
+                # In that case, it will not be available here because we are still in the main process.
+                # but instead should be populated remotely before calling the task_function.
+                sweep_config.hydra.job.id = "job_id_for_{}".format(idx)
+                sweep_config.hydra.job.num = idx
+            HydraConfig.instance().set_config(sweep_config)
+            singleton_state = Singleton.get_state()
+            self._dump_func_params(
+                idx,
+                sweep_config,
+                self.config,
+                self.task_function,
+                singleton_state,
+                temp_dir_job_dump,
+            )
+
+        remote_pickle_dir = self._get_remote_dir(ray_yaml_path)
+        self._rsync_to_ray_cluster(
+            os.path.join(temp_dir_job_dump, ""), remote_pickle_dir, ray_yaml_path
+        )
+
+        self._rsync_to_ray_cluster(
+            os.path.join(os.path.dirname(__file__), "ray_remote_invoke.py"),
+            remote_script_dir,
+            ray_yaml_path,
+        )
+
+        self._invoke_functions_on_ray(
+            ray_yaml_path,
+            os.path.join(remote_script_dir, "ray_remote_invoke.py"),
+            remote_pickle_dir,
+        )
+        # the script running on Ray cluster cannot pass result back to our local machine.
+        return []
+
+    def _invoke_functions_on_ray(
+        self, yaml_file_path: str, file_path: str, pickle_path: str
+    ):
+        command = "python {} {}".format(file_path, pickle_path)
+        log.info("Issuing command on remote server: {}".format(command))
+        ray_proc = Popen(
```

Comment:
I think you might need a mechanism to pass additional options to `ray exec`, see: https://ray.readthedocs.io/en/latest/package-ref.html#ray-exec

For example, when a docker container is used, `ray exec --docker CLUSTER_CONFIG_FILE python ...` should be called.